### PR TITLE
NGDOC-240 Document writeable variables

### PR DESCRIPTION
**Pull request by nibinqtl**

### CODE REVIEW

**Review by Jianghai-Quantil — CHANGES_REQUESTED** [review]
The variable $cache_misc is also writable.

**Comment by nibinqtl** [comment]
> The variable $cache_misc is also writable.

Good point, just updated.

File: en/docs/edge-logic/built-in-variables.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Built-in Variables
 
-The table below lists all the built-in variables supported by the CDN Pro servers. You can use them in Edge Logic or [Load Balancer Logic](lb7-es-structure). Not all variables are supported in both places. The [real-time log](/docs/portal/edge-configurations/creating-property#real-time-log) supports the same set of variables as the Load Balancer Logic. The variables with numerical values are tagged with <span class="badge small" title="numerical value">#</span>.
+The table below lists all the built-in variables supported by the CDN Pro servers. You can use them in Edge Logic or [Load Balancer Logic](lb7-es-structure). Not all variables are supported in both places. The [real-time log](/docs/portal/edge-configurations/creating-property#real-time-log) supports the same set of variables as the Load Balancer Logic. Most variables here are read-only and the writable ones are indicated with <span class="badge green">R/W</span>. The variables with numerical values are tagged with <span class="badge small" title="numerical value">#</span>.
```
Comment (burtonquantil) [inline en/docs/edge-logic/built-in-variables.md:3]: I'd write   "Most variables here are read-only. The writeable ones are indicated by <span class="badge green">R/W</span."

File: en/docs/edge-logic/built-in-variables.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ If you use the CDN Pro portal to edit the property, typing `$` at a position whe
...
 | <span id="random_N"></span><span class="var">$random_N</span> |<span class="badge small" title="numerical value">#</span> A random integer in [0, N-1]. N can be an integer in [2,1e9].| <span class="badge green">Yes</span> | <span class="badge green">Yes</span> |
-| <span id="realtime_log_ds_factor"></span><span class="var">$realtime_log_ds_factor</span> |<span class="badge small" title="numerical value">#</span> RT log downsample factor. A value N means one log entry for every N requests| <span class="badge yellow">No</span> | <span class="badge green">Yes</span> |
+| <span id="realtime_log_ds_factor"></span><span class="var">$realtime_log_ds_factor</span> |<span class="badge small" title="numerical value">#</span> RT log downsample factor. A value N means one log entry for every N requests. Users can write to this variable to [override the default sample rate](/docs/edge-logic/supported-directives#realtime_log_downsample). | <span class="badge yellow">No</span> | <span class="badge green">R/W</span> |
```
Comment (burtonquantil) [inline en/docs/edge-logic/built-in-variables.md:47]: I'd change "Users can write to this variable to ..."  to   "Users can set a value.... "

File: en/docs/edge-logic/supported-directives.md [hunk 3]
```diff
@@ -1039,7 +1039,7 @@ This directive can be used to instruct the proxy server to reorder and merge mul
...
+Overrides the main "Sample Rate" specified for the [Real-Time Log](/docs/portal/edge-configurations/creating-property#real-time-log). `factor` can be an integer in [0, 65535] or a variable. A value of 0 disables the logging; 1 means do not downsample; N>1 means one log entry for every N requests. If the variable value is an empty string, the main setting is not overridden. An invalid value in the variable results in a factor of 100. The actual sample factor can be included in the log with the variable [`$realtime_log_ds_factor`](/docs/edge-logic/built-in-variables#realtime_log_ds_factor). This directive can only be used in the Edge Logic. In the Load Balancer Logic, one can directly use `set $realtime_log_ds_factor {factor}` to override the default sample rate.
```
Comment (burtonquantil) [inline en/docs/edge-logic/supported-directives.md:1042]: What does "main" mean in this context"?    "Main" implies there is a secondary or other settings.

**Review by burtonquantil — CHANGES_REQUESTED** [review]
Looks pretty good. I made a few text suggestions.

File: en/docs/edge-logic/built-in-variables.md [hunk 4]
```diff
@@ -44,7 +44,7 @@ If you use the CDN Pro portal to edit the property, typing `$` at a position whe
...
 | <span id="random_N"></span><span class="var">$random_N</span> |<span class="badge small" title="numerical value">#</span> A random integer in [0, N-1]. N can be an integer in [2,1e9].| <span class="badge green">Yes</span> | <span class="badge green">Yes</span> |
-| <span id="realtime_log_ds_factor"></span><span class="var">$realtime_log_ds_factor</span> |<span class="badge small" title="numerical value">#</span> RT log downsample factor. A value N means one log entry for every N requests| <span class="badge yellow">No</span> | <span class="badge green">Yes</span> |
+| <span id="realtime_log_ds_factor"></span><span class="var">$realtime_log_ds_factor</span> |<span class="badge small" title="numerical value">#</span> RT log downsample factor. A value N means one log entry for every N requests. Users can set value to [override the default sample rate](/docs/edge-logic/supported-directives#realtime_log_downsample). | <span class="badge yellow">No</span> | <span class="badge green">R/W</span> |
```
Comment (burtonquantil) [inline en/docs/edge-logic/built-in-variables.md:47]: you should write:  "Users can set a value"

File: en/docs/edge-logic/supported-directives.md [hunk 5]
```diff
@@ -989,7 +989,7 @@ A restriction of this directive is that it works only when body size is less tha
 This directive assigns the `value` to the `$variable`. The `value` can be another variable or a composition of variables and literals. While this directive looks very similar to the [`set`](#set) directive, it differs in when it is executed. The `set` directive is executed during the "rewrite" phases which are very early -- almost right after the request is received from the client. On the contrary, `proxy_set` is executed after the response header is received from the origin (in case of a cache miss) or read from the cache. Therefore, the `value` can have information contained in the response header (after being modified by any [`origin_header_modify`](#origin_header_modify) directive). In addition, this directive supports the `if()` parameter which can set a condition for the assignment to happen. Here are a few examples:
 ```nginx
 set $cache_time 1d; # by default, cache for 1 day
-# if origin responds with a "cachetime" header, use it to override the default
+# if origin responds with a "cachetime" header, use it to  the default
```
Comment (burtonquantil) [inline en/docs/edge-logic/supported-directives.md:992]: Did you accidentally delete the word 'override', or were you intending to replace it with something else?

**Review by burtonquantil — CHANGES_REQUESTED** [review]
made a couple of suggestions.

File: en/docs/edge-logic/built-in-variables.md [hunk 6]
```diff
@@ -9,15 +9,15 @@ If you use the CDN Pro portal to edit the property, typing `$` at a position whe
...
 | <span id="client_province_code"></span><span class="var">$client_province_code</span> | client's China province code | <span class="badge green">Yes</span> | <span class="badge green">Yes</span> |
 | <span id="client_real_ip"></span><span class="var">$client_real_ip</span> | client's IP address | <span class="badge green">Yes</span> | <span class="badge green">Yes</span> |
 | <span id="connection_requests"></span><span class="var">$connection_requests</span> |<span class="badge small" title="numerical value">#</span> current number of requests in the connection | <span class="badge yellow">No</span> | <span class="badge green">Yes</span> |
-| <span id="content_code"></span><span class="var">$content_code</span> | a code assigned to content for reporting and billing purposes | <span class="badge green">Yes</span> | <span class="badge green">Yes</span> |
+| <span id="content_code"></span><span class="var">$content_code</span> | a writable variable to [assign code to content for reporting and billing purposes](/docs/edge-logic/faq#how-to-classify-traffic-in-reports-based-on-custom-rules) | <span class="badge green">R/W</span> | <span class="badge green">R/W</span> |
```
Comment (burtonquantil) [inline en/docs/edge-logic/built-in-variables.md:20]: "assign code"  would be better as "assign a code"

**Review by burtonquantil — APPROVED** [review]
OK to merge.. I made one small suggestion.

File: en/docs/edge-logic/built-in-variables.md (hunk 6 above)
Comment (nibinqtl) [reply en/docs/edge-logic/built-in-variables.md]: OK, fixed. In the future, for English grammatical problems, you can directly update the code to save some time and efforts.

**Review by burtonquantil — APPROVED** [review]
OK to merge

**Comment by nibinqtl** [comment]
@Jianghai-Quantil , still needs your approval.

**Review by Jianghai-Quantil — APPROVED** [review]
OK to merge.

**Pull request merged**